source/MRViewer/MRRenderMeshObject.cpp

```diff
@@ -356,7 +352,7 @@ void RenderMeshObject::bindMesh_( bool alphaSort ) const
         GL_EXEC( glPixelStorei( GL_UNPACK_ALIGNMENT, 1 ) );
 
         auto facesColorMap = objMesh_->getFacesColorMap();
-        auto res = calcTextureRes( int( facesColorMap.size() ), maxTexSize );
+        auto res = calcTextureRes( int( facesColorMap.size() ), maxTexSize_ );
```

Comment:
can we calc texture resolution only in `update_` and not calc it twice

Reply:
`facesColorMap` is pre-calculated outside this class.